registry/deploying.md

```diff
@@ -1,94 +1,246 @@
+  --name registry \
+  -v `pwd`/data:/var/lib/registry \
+  registry:2
+```
 
-### Alternatives
+### Customize the storage back-end
 
-You should usually consider using [another storage backend](./storage-drivers/index.md) instead of the local filesystem. Use the [storage configuration options](./configuration.md#storage) to configure an alternate storage backend.
+Instead of the local filesystem, you can use a different storage back-end by
+making use of [storage drivers](./storage-drivers/index.md). For instance, you
+can store your registry in an Amazon S3 bucket or on Google Cloud Platform. For
+more information, see
+[storage configuration options](./configuration.md#storage).
 
-Using one of these will allow you to more easily scale your registry, and leverage your storage redundancy and availability features.
 
-## Running a domain registry
+## Run an externally-accessible registry
 
-While running on `localhost` has its uses, most people want their registry to be more widely available. To do so, the Docker engine requires you to secure it using TLS, which is conceptually very similar to configuring your web server with SSL.
+Running a registry only accessible on `localhost` has limited usefulness. In
+order to make your registry accessible to external hosts, you must first secure
+it using TLS.
 
 ### Get a certificate
 
-Assuming that you own the domain `myregistrydomain.com`, and that its DNS record points to the host where you are running your registry, you first need to get a certificate from a CA.
+These examples assume the following:
 
-Create a `certs` directory:
+- Your registry will be accessible on `https://myregistry.domain.com/`.
+- Your DNS, routing, and firewall settings allow access to the registry's host
+  on port 5000.
+- You have already obtained a certificate from a certificate authority (CA).
 
-    mkdir -p certs
+If you have been issued an _intermediate_ certificate instead, see
+[use an intermediate certificate](#use-an-intermediate-certificate).
 
-Then move and/or rename your crt file to: `certs/domain.crt`, and your key file to: `certs/domain.key`.
+1.  Create a `certs` directory.
 
-Make sure you stopped your registry from the previous steps, then start your registry again with TLS enabled:
+    ```bash
+    $ mkdir -p certs
+    ```
 
-    docker run -d -p 5000:5000 --restart=always --name registry \
+    Copy the `.crt` and `.key` files from the CA into the `certs` directory.
+    The following steps assume that the files are named `domain.crt` and
+    `domain.key`.
+
+2.  Stop the registry if it is currently running.
+
+    ```bash
+    $ docker stop registry
+    ```
+
+3.  Restart the registry, directing it to use the TLS certificate. This command
+    bind-mounts the `certs/` directory into the container at `/certs/`, and sets
+    environment variables that tell the container where to find the `domain.crt`
+    and `domain.key` file.
+
+    ```bash
+    $ docker run -d \
+      -p 5000:5000 \
```

Comment:
Can we do this example without port 5000? The section above makes that promise then the example breaks that promise. ;)